src/Simulation/Simulators/CommonNativeSimulator/Dump.cs

```diff
@@ -96,5 +92,393 @@ public QSimDumpRegister(CommonNativeSimulator m) : base(m)
+                {
+                    return "                   ";
+                }
+
+                var chart = "    ---";
+                if (angle > 0)
+                {
+                    if (angle >= (0 * PI / 8) + offset && angle < ((1 * PI / 8) + offset)) { chart = "     /-"; }
+                    if (angle >= (1 * PI / 8) + offset && angle < ((2 * PI / 8) + offset)) { chart = "     / "; }
+                    if (angle >= (2 * PI / 8) + offset && angle < ((3 * PI / 8) + offset)) { chart = "    +/ "; }
+                    if (angle >= (3 * PI / 8) + offset && angle < ((4 * PI / 8) + offset)) { chart = "   ↑   "; }
+                    if (angle >= (4 * PI / 8) + offset && angle < ((5 * PI / 8) + offset)) { chart = " \\-    "; }
+                    if (angle >= (5 * PI / 8) + offset && angle < ((6 * PI / 8) + offset)) { chart = " \\     "; }
+                    if (angle >= (6 * PI / 8) + offset && angle < ((7 * PI / 8) + offset)) { chart = "+\\     "; }
+                    if (angle >= (7 * PI / 8) + offset) { chart = "---    "; }
+                }
+                else  if (angle < 0)
+                {
+                    var abs_angle = System.Math.Abs(angle);
+                    if (abs_angle >= (0 * PI / 8) + offset && abs_angle < ((1 * PI / 8) + offset)) { chart = "     \\+"; }
+                    if (abs_angle >= (1 * PI / 8) + offset && abs_angle < ((2 * PI / 8) + offset)) { chart = "     \\ "; }
+                    if (abs_angle >= (2 * PI / 8) + offset && abs_angle < ((3 * PI / 8) + offset)) { chart = "    -\\ "; }
+                    if (abs_angle >= (3 * PI / 8) + offset && abs_angle < ((4 * PI / 8) + offset)) { chart = "   ↓   "; }
+                    if (abs_angle >= (4 * PI / 8) + offset && abs_angle < ((5 * PI / 8) + offset)) { chart = " /+    "; }
+                    if (abs_angle >= (5 * PI / 8) + offset && abs_angle < ((6 * PI / 8) + offset)) { chart = " /     "; }
+                    if (abs_angle >= (6 * PI / 8) + offset && abs_angle < ((7 * PI / 8) + offset)) { chart = "-/     "; }
+                }
+
+                return $" {chart} [ {angle,8:F5} rad ]";
+            }
+
+            /// <summary>
+            /// Returns a string for the amplitude's polar representation (magnitude/angle).
+            /// </summary>
+            public virtual string FormatPolar(double magnitude, double angle) =>
+                $"{FormatMagnitude(magnitude, angle)}{FormatAngle(magnitude, angle)}";
+
+            /// <summary>
+            /// Returns a string for the amplitude's cartesian representation (real + imagnary).
+            /// </summary>
+            public virtual string FormatCartesian(double real, double img) =>
+                $"{real,9:F6} + {img,9:F6} i";
+
+            /// <summary>
+            /// The method to use to format the amplitude into a string.
+            /// </summary>
+            public virtual string Format(uint idx, double real, double img)
```

Comment:
Is this method still used? It looks like it's been superseded by `ToString` below?

Reply:
I'm removing `Format()` and I assume that other notes about changing the signature or structure of the `Format()` are not applicable any more.